sdk/appconfiguration/app-configuration/src/internal/tracingHelpers.ts

```diff
@@ -1,100 +1,54 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-import { getTracer } from "@azure/core-tracing";
-import { Span, SpanKind, CanonicalCode } from "@opentelemetry/api";
-import { SpanOptions } from "@azure/core-tracing";
+import { Span, StatusCode } from "@opentelemetry/api";
 
-import { RestError } from "@azure/core-http";
+import { RestError, createSpanFunction, OperationOptions } from "@azure/core-http";
+import { AppConfigurationClient } from "../appConfigurationClient";
 
-/**
- * @internal
- */
-export interface Spannable {
-  spanOptions?: SpanOptions;
-}
+/** @internal */
+export const createSpan = createSpanFunction({
+  namespace: "Microsoft.AppConfiguration",
+  packagePrefix: "Azure.Data.AppConfiguration"
+});
 
 /**
+ * Traces an operation and properly handles reporting start, end and errors for a given span
+ *
+ * @param operationName - Name of a method in the TClient type
+ * @param options - An options class, typically derived from \@azure/core-http/RequestOptionsBase
+ * @param fn - The function to call with an options class that properly propagates the span context
+ *
  * @internal
  */
-export class Spanner<TClient> {
-  constructor(private baseOperationName: string) {}
-
-  /**
-   * Traces an operation and properly handles reporting start, end and errors for a given span
-   *
-   * @param operationName - Name of a method in the TClient type
-   * @param options - An options class, typically derived from \@azure/core-http/RequestOptionsBase
-   * @param fn - The function to call with an options class that properly propagates the span context
-   * @param translateToCanonicalCodeFn - An optional function to translate thrown errors into a CanonicalCode for the span
-   */
-  async trace<OptionsT extends Spannable, ReturnT>(
-    operationName: keyof TClient,
-    options: OptionsT,
-    fn: (options: OptionsT, span: Span) => Promise<ReturnT>,
-    translateToCanonicalCodeFn: (err: Error) => CanonicalCode = Spanner.getCanonicalCode
-  ): Promise<ReturnT> {
-    const { newOptions, span } = this.createSpan<OptionsT>(options, operationName);
-
-    try {
-      return await fn(newOptions, span);
-    } catch (err) {
-      span.setStatus({
-        code: translateToCanonicalCodeFn(err),
-        message: err.message
-      });
-      throw err;
-    } finally {
-      span.end();
-    }
-  }
-
-  private createSpan<T extends Spannable>(options: T, operationName: keyof TClient) {
-    const span = getTracer().startSpan(`${this.baseOperationName}.${operationName}`, {
-      ...options.spanOptions,
-      kind: SpanKind.INTERNAL
+export async function trace<ReturnT>(
+  operationName: keyof AppConfigurationClient,
+  options: OperationOptions,
+  fn: (options: OperationOptions, span: Span) => Promise<ReturnT>,
+  createSpanFn = createSpan
```

Comment:
nit: maybe call this arg `createSpan` and have the default implementation be something like `createSpanImpl` or `createSpanDefault` ?